sklearn/tree/_tree.pyx

```diff
@@ -920,9 +1110,36 @@ cdef inline void _init_split(SplitRecord* self, SIZE_t start_pos) nogil:
 
 
 cdef class Splitter:
+    """Abstract splitter class.
+
+    Splitters are called by tree builders to find the best splits on both
+    sparse and dense data, one split at a time.
+    """
+
     def __cinit__(self, Criterion criterion, SIZE_t max_features,
                   SIZE_t min_samples_leaf, double min_weight_leaf,
                   object random_state):
+        """
+        Parameters
+        ----------
+        criterion: Criterion
+            The criterion to measure the quality of a split. 
+
+        max_features: SIZE_t
+            The maximal number of features that a single decision tree can
+            consider.
+
+        min_samples_leaf: SIZE_t
+            The minimal number of samples each leaf can have.
```

Comment:
Do not consider splits that would results in less training samples than ``min_samples_leaf`` after the split.